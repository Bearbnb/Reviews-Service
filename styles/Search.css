.searchDiv {
    display: grid;
    grid-template-columns: 1fr 20fr;
    position: relative !important;
    height: 100% !important;
    width: 100% !important;
    border: 1px solid #e1e2e3;
    border-radius: 2px
}

.searchContainer {
    display: inline-block !important;
    vertical-align: bottom !important;

}
#searchInput {
    background-color: transparent !important;
    border: 0px !important;
    width: 100% !important;
    font-size: 14px !important;
    line-height: 18px !important;
    letter-spacing: normal !important;
    /* font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif !important; */
    font-weight: 600 !important;
    padding-top: 7px !important;
    padding-bottom: 7px !important;
}
input {
    outline: none;
}
#searchIcon {
    float: left;
    margin-left: 10px;
    margin-top: 5px;

}
