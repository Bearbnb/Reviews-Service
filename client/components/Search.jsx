import React from 'react';
import ReactDOM from 'react-dom';
import styles from '../../styles/Search.css';


class Search extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      term: '',
    };
  }


  onChange(e) {
    this.setState({
      term: e.target.value,
    });
  }

  handleKeyPress(e) {
    const searchFunc = this.props.onClick;
    if (e.key === 'Enter') {
      e.target.value = '';
      searchFunc(this.state.term);
    }
  }

  render() {
    return (
      <div className={styles.searchDiv}>
        <div id={styles.searchIcon}>
          <i className="fas fa-search" />
        </div>
        <div className={styles.searchContainer}>
          <input
            id={styles.searchInput}
            value={this.state.name}
            placeholder="Search reviews:"
            onChange={this.onChange.bind(this)}
            onKeyPress={e => this.handleKeyPress(e)}
          />
        </div>
      </div>
    );
  }
}

export default Search;
