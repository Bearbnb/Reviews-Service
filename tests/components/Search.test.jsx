import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import Search from '../../client/components/Search';

describe('<Search />', () => {
  it('should render properly', () => {
    const wrapper = shallow(<Search />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should have a state', () => {
    const wrapper = shallow(<Search />);
    expect(wrapper.state().term).toEqual(null);
  });
  // it('should handle press key on enter input', () => {
  //   const onChange = jest.fn();
  //   const wrapper = shallow(<Search onChange={onChange} />);
  //   expect(wrapper.find('div')).to.equal(true);
  // });
});
