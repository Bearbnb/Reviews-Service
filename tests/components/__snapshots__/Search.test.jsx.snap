// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> should render properly 1`] = `
<div
  className="searchDiv"
>
  <div
    id="searchIcon"
  >
    <i
      className="fas fa-search"
    />
  </div>
  <div
    className="searchContainer"
  >
    <input
      id="searchInput"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Search reviews:"
    />
  </div>
</div>
`;
